js-packages/@prestojs/viewmodel/src/__tests__/ViewModelCache.test.ts

```diff
@@ -34,23 +29,19 @@ test('should cache records from record instance', () => {
 });
 
 test('should cache records from plain object', () => {
-    class Test1 extends ViewModel {
-        static _fields = {
-            id: F('id'),
-        };
-    }
-    class Test2 extends ViewModel {
-        static _fields = {
-            id: F('id'),
-        };
-    }
+    class Test1 extends ViewModelFactory({
+        id: F('id'),
+    }) {}
+    class Test2 extends ViewModelFactory({
+        id: F('id'),
+    }) {}
 
     const record1Data = { id: 5 };
 
     const record1 = Test1.cache.add(record1Data);
 
     expect(record1).toBeInstanceOf(Test1);
-    expect(record1 instanceof ViewModel ? record1.toJS() : false).toEqual(record1Data);
+    expect(isViewModelInstance(record1) ? record1.toJS() : false).toEqual(record1Data);
```

Comment:
The only possibility would be to make the factory class itself the base... maybe.

I would personally find that a bit confusing knowing the the factory function itself also serves as the base class but maybe...? what do you think?

Reply:
Yeah, if that makes `instanceof` work it would be worth it.

I expect that someone looking at the library code would probably think "wtf is going on here" but as long as you add a `// comment` around it explaining why we do it that way then that's a net positive

Reply:
To clarify, my understanding is that you would end up with:

- `ViewModelInterface` -- typescript definition
- `ViewModel` -- a base class (may be a placeholder that doesn't really do much)
- `ViewModelFactory` -- a class generator that returns a class that is a descendant of ViewModel